packages/builders/src/messages/embed/UnsafeEmbed.ts

```diff
@@ -138,7 +182,7 @@ export class UnsafeEmbed implements APIEmbed {
 	 * @param fields The replacing field objects
 	 */
 	public spliceFields(index: number, deleteCount: number, ...fields: APIEmbedField[]): this {
-		this.fields.splice(index, deleteCount, ...Embed.normalizeFields(...fields));
+		this.fields?.splice(index, deleteCount, ...Embed.normalizeFields(...fields));
```

Comment:
`data` property + `UnsafeEmbed` instead of `Embed`
```suggestion
		this.data.fields?.splice(index, deleteCount, ...UnsafeEmbed.normalizeFields(...fields));
```